src/poetry/core/packages/dependency.py

```diff
@@ -558,7 +554,7 @@ def create_from_pep_508(
                 name = m.group("name")
                 version = m.group("ver")
 
-            name = req.name or link.egg_fragment
+            name = cast(str, req.name or link.egg_fragment)
```

Comment:
it's not clear what the intention of the (original) code is here: the name extracted only three lines previously is discarded here.  So something has gone wrong and it would be good to take the opportunity to fix it.

My guess is that this whole fragment can be discarded (anyway that doesn't break any tests).

Reply:
(and if that's right then there are no users of `egg_fragment` left so that can be discarded too)